[Brewmaster] adding spinning crane kick usage with charred passions legendary equipped

rbatche opened the pull request:
added spinning crane kick for ST if charred passions legendary is equipped + charred passions buff up

rbatche committed:
[Brewmaster] add spinning crane kick with charred passions legendary

added spinning crane kick for ST if charred passions legendary is equipped + charred passions buff up

rbatche commented:
there probably should be an energy check here for spinning crane kick, but i'm not quite sure how to do that.

mrdmnd commented:
I think S.SpinningCraneKick:IsCastable() gets that for you.

This looks good to me. Did you test it okay locally?

Cilraaz commented:
S.SpinningCraneKick:IsReady() would check IsCastable() and IsUsable(), which would be the check for if it's available and if the player has enough energy.

mrdmnd commented:
Sounds like an easy fix. If you want to push a new commit, I'll happily merge.

rbatche committed:
[Brewmaster] more changes for charred passions legendary

added some, 'logic'?, if using this legendary, it favours using BK on CD, with SCK and not weaving in KegSmash unless shuffle is down and enemy count is 1

added something to prevent BK being used right before a breath of fire cast if using charred passions and shuffle is already up (using BK on CD with the legendary buff up really shines over just spamming SCK)

now only displaying SCK when energy is available

weapons of order is now only suggested after keg smash is on CD (usually this would be a single cast of KS on pull, WoD then KS again)

rbatche commented:
thanks for the help. have made the change + some additional changes to get the most dmg output out of charred passions while still maintaining the shuffle defensive buff.

Cilraaz commented on HeroRotation_Monk/Brewmaster.lua:
```diff
@@ -173,8 +173,8 @@ local function APL()
-      -- weapons_of_order
-      if S.WeaponsOfOrder:IsCastable() then
+      -- weapons_of_order (if KegSmash is on CD)
+      if S.WeaponsOfOrder:IsCastable() and S.KegSmash:CooldownRemains() then
```
CooldownRemains() should be checked against a value. In this case, it'd be better to use "not S.KegSmash:CooldownUp()". This should function properly as is, but CooldownUp is typically used in situations like these in other profiles.

rbatche replied:
changed in commit 7654303afe03a1ca77b0fba1a5e01fcd322c6b4e

Cilraaz commented on HeroRotation_Monk/Brewmaster.lua:
```diff
@@ -216,8 +216,20 @@ local function APL()
+      if HR.Cast(S.BreathOfFire, nil, nil, not Target:IsInMeleeRange(8)) then return "Breath of Fire 2"; end
+    end
+	-- blackout_strike>KegSmash if only 1 target, and shuffle buff is up & charred passion buff up
+    if S.KegSmash:IsCastable() and S.BlackoutKick:IsCastable() and Player:BuffUp(S.CharredPassions) and Player:BuffUp(S.Shuffle) and EnemiesCount8 <= 1 then
```
Spells with a cost (mana, energy, chi, etc) should use IsReady(), rather than IsCastable(), as IsCastable() will return true even when the spell is not usable due to resources. Also, conditions aside from the initial IsCastable/IsReady check should be grouped into their own parentheses. This isn't required, but helps segregate conditionals.

rbatche replied:
updated some spells to IsReady & parenthesised conditionals in commit 7654303afe03a1ca77b0fba1a5e01fcd322c6b4e

Cilraaz commented on HeroRotation_Monk/Brewmaster.lua:
```diff
@@ -235,6 +235,14 @@ local function APL()
+    end
+    if Player:HasLegendaryEquipped(86) and S.BlackoutKick:IsCastable() and Player:BuffUp(S.CharredPassions) then
+      if HR.Cast(S.BlackoutKick, nil, nil, not Target:IsSpellInRange(S.BlackoutKick)) then return "Blackout Kick"; end
+    end
```
Indenting got a bit out of whack here. Also, IsReady should be used for spells with a cost.

rbatche replied:
fixed indenting in commit 0d610acc607049ff910e5bd0c653c182049a1788

rbatche committed:
[Brewmaster] changes based on Cilraaz's comments

ensuring KS is a priority cast without charred passions legendary

ensuring KS is used during charred passions buff if BK has already been used

ensuring KS is still used as a priority if the charred passions buff is not up but legendary is being used

added usage of BK where the CD would be available again after a BoF cast

added a CD check during charred passions buff to prevent too many casts of SCK when a BK is going to be available during that channel

rbatche committed:
[Brewmaster] fixing some indenting

missed previous comment about indenting & fixed

mrdmnd commented:
Looks good to me; thank you for your contribution!
